Add Next.js server action instrumentation functionality (#8488)
getsentry/sentry-docs: src/platforms/javascript/guides/nextjs/manual-setup.mdx

// export default withSentryConfig(nextConfig, sentryWebpackPluginOptions);
```

## Instrument Next.js Server Actions

_Requires `@sentry/nextjs` SDK version `7.81.0` or newer._

> To instrument Next.js server actions, wrap their content in `withServerActionInstrumentation`, along with a name to describe your server action.

Reviewer: It looks like "Server Actions" are always capitalized, from a quick google.

```suggestion
To instrument Next.js Server Actions, wrap them in `withServerActionInstrumentation`, along with a name to describe your server action.
```

> You can optionally pass form data, headers to record them and configure the wrapper to record the Server Action responses:

Reviewer: ```suggestion
You can optionally pass form data and headers to record them, and configure the wrapper to record the Server Action responses:
```

```tsx
import * as Sentry from "@sentry/nextjs";
import { headers } from "next/headers";

export default function ServerComponent() {
  async function myServerAction(formData: FormData) {
    "use server";
    return await Sentry.withServerActionInstrumentation(
      "myServerAction", // The name you want to associate this Server Action with in Sentry
      {
        formData, // Optionally pass in the form data
        headers: headers(), // Optionally pass in headers
        recordResponse: true, // Optionally record the server action response
      },
      async () => {

>         // ... Your server action code

Reviewer: ```suggestion
        // ... Your Server Action code
```
